### PR TITLE
ci: Migrate windows conda to GHA

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73393
* __->__ #73262

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
Somehow this PR caused https://hud.pytorch.org/ci/pytorch/pytorch/master?name_filter=windows-binary to start timing out

**Review by atalman — APPROVED** [review]

File: .circleci/scripts/binary_windows_build.sh [hunk 1]
```diff
@@ -71,6 +72,7 @@ pushd "$BUILDER_ROOT"
 if [[ "$PACKAGE_TYPE" == 'conda' ]]; then
     ./windows/internal/build_conda.bat
 elif [[ "$PACKAGE_TYPE" == 'wheel' || "$PACKAGE_TYPE" == 'libtorch' ]]; then
+    export NIGHTLIES_PYTORCH_ROOT="$PYTORCH_ROOT"
```
Comment (seemethere) [inline .circleci/scripts/binary_windows_build.sh:75]: Adding this here should fix timeout issues with libtorch/manywheel builds

**Pull request closed without merging**